src/crypto/tls/metrics_test.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2020 Cloudflare, Inc. All rights reserved. Use of this source code
+// is governed by a BSD-style license that can be found in the LICENSE file.
+
+package tls
+
+import (
+	"crypto/x509"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"testing"
+	"time"
+)
+
+func (e EXP_EventTLS13ServerHandshakeTimingInfo) isMonotonic() bool {
+	return e.ReadClientHello <= e.WriteServerHello &&
+		e.WriteServerHello <= e.WriteEncryptedExtensions &&
+		e.WriteEncryptedExtensions <= e.WriteCertificate &&
+		e.WriteCertificate <= e.WriteCertificateVerify &&
+		e.WriteCertificateVerify <= e.WriteServerFinished &&
+		e.WriteServerFinished <= e.ReadClientFinished
+}
+
+func (e EXP_EventTLS13ClientHandshakeTimingInfo) isMonotonic() bool {
+	return e.WriteClientHello <= e.ReadServerHello &&
+		e.ReadServerHello <= e.ReadEncryptedExtensions &&
+		e.ReadEncryptedExtensions <= e.ReadCertificate &&
+		e.ReadCertificate <= e.ReadCertificateVerify &&
+		e.ReadCertificateVerify <= e.ReadServerFinished &&
+		e.ReadServerFinished <= e.WriteClientFinished
+}
+
+type timingInfo struct {
```

Comment:
I would name this something that reflects it's only meant to be used for testing, e.g., `testTimingResult` or something.

Reply:
👍